core/src/main/scala/kafka/coordinator/TransactionMetadata.scala

```diff
@@ -0,0 +1,104 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one or more
+  * contributor license agreements.  See the NOTICE file distributed with
+  * this work for additional information regarding copyright ownership.
+  * The ASF licenses this file to You under the Apache License, Version 2.0
+  * (the "License"); you may not use this file except in compliance with
+  * the License.  You may obtain a copy of the License at
+  *
+  *    http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+package kafka.coordinator
+
+import kafka.utils.nonthreadsafe
+import org.apache.kafka.common.TopicPartition
+
+import scala.collection.mutable
+
+private[coordinator] sealed trait TransactionState { def byte: Byte }
+
+/**
+  * Transaction has not existed yet
+  *
+  * transition: received AddPartitionsToTxnRequest => Ongoing
+  *             received AddOffsetsToTxnRequest => Ongoing
+  */
+private[coordinator] case object NotExist extends TransactionState { val byte: Byte = 0 }
```

Comment:
This is probably a good reason to use separate packages. We don't want the namespace to get too convoluted.